### PR TITLE
Fix bulkCreate with updateOnDuplicate option, Now it properly maps fields to attributes from camelCase to snake_case

### LINKED ISSUE #9016: BulkCreate: updateOnDuplicate option is not mapping fields correctly
## What are you doing?

Hi guys,
I have found that bulkCreate with updateOnDuplicate option is not mapping fields correctly when I use snake-case in the mysql structure and camel-case in model definitions.

Model:
```js
const Model = sequelize.define('userCallUp', {
        id: {
            type: DataTypes.INTEGER(10).UNSIGNED,
            allowNull: false,
            primaryKey: true,
            autoIncrement: true,
            field: 'id'
        },
        idCallUp: {
            type: DataTypes.INTEGER(10).UNSIGNED,
            allowNull: false,
            references: {
                model: 'callUp',
                key: 'id'
            },
            field: 'id_call_up'
        },
        idUser: {
            type: DataTypes.INTEGER(10).UNSIGNED,
            allowNull: true,
            references: {
                model: 'user',
                key: 'id'
            },
            field: 'id_user'
        },
        number: {
            type: DataTypes.INTEGER(3).UNSIGNED,
            allowNull: true,
            field: 'number'
        },
        nickname: {
            type: DataTypes.CHAR(255),
            allowNull: true,
            field: 'nickname'
        },
        group: {
            type: DataTypes.ENUM('bench','pitch','coachs'),
            allowNull: false,
            defaultValue: 'bench',
            field: 'group'
        },
        createdAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'created_at'
        },
        updatedAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'updated_at'
        },
        deletedAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'deleted_at'
        }
    }, {
        tableName: 'user_call_up'
    });
```

When calling:
```
const usersCallsUp = [{
                        "idCallUp": 21,
                        "idUser": 500,
                        "number": null,
                        "nickname": null,
                        "group": "coachs"
                    },{
                        "idCallUp": 21,
                        "idUser": 501,
                        "number": 1,
                        "nickname": null,
                        "group": "pitch"
                    }];
UserCallUp.bulkCreate(usersCallsUp, {updateOnDuplicate: ['idUser', 'number', 'nickname', 'group']});
```
## What do you expect to happen?
I expect that all fields that are including in the updateOnDuplicate option were mapping propertly on ON DUPLICATE KEY UPDATE clause of SQL query.
Expected SQL query:
```
INSERT INTO `user_call_up` (`id`,`id_call_up`,`id_user`,`group`,`created_at`,`updated_at`,`number`) VALUES (NULL,21,500,'coachs','2018-02-08 17:14:09','2018-02-08 17:14:09',NULL),(NULL,21,501,'pitch','2018-02-08 17:14:09','2018-02-08 17:14:09',1) ON DUPLICATE KEY UPDATE `id_user`=VALUES(`id_user`),`number`=VALUES(`number`),`nickname`=VALUES(`nickname`),`group`=VALUES(`group`);
```

## What is actually happening?
A sequelize error is raised because on ON DUPLICATE KEY UPDATE the field "idUser" is not mapping correctly to "id_user" which is the correct field on database. (camelCase to snake_case)
Sequelize error:
```
{ SequelizeDatabaseError: Unknown column 'idUser' in 'field list' ... }
```
SQL query:
```
INSERT INTO `user_call_up`
(`id`,`id_call_up`,`id_user`,`group`,`created_at`,`updated_at`,`number`) VALUES (NULL,21,500,'coachs','2018-02-08 17:14:09','2018-02-08 17:14:09',NULL),(NULL,21,501,'pitch','2018-02-08 17:14:09','2018-02-08 17:14:09',1) ON DUPLICATE KEY UPDATE `idUser`=VALUES(`idUser`),`number`=VALUES(`number`),`nickname`=VALUES(`nickname`),`group`=VALUES(`group`);
```
Also, if you include the value of database field "id_user" into the array like that, ['id_user', 'number', 'nickname', 'group'], sequelize ignore this field to the clause.

__Dialect:__ mysql
__Dialect version:__ 2.11.1
__Database version:__ 5.7.18
__Sequelize version:__ 4.33.0
__Tested with latest release:__ 4.33.0

Thanks!
Manu Rua

### PR DESCRIPTION
**Pull request by ManuRua**
### Pull Request check-list

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](../CONTRIBUTING.md)?

### Description of change
Closes #9016 

Now fields with camelCase on updateOnDuplicate array will be properly maps to attributes with snake_case.

### CODE REVIEW

File: lib/dialects/abstract/query-generator.js [hunk 1]
```diff
@@ -249,8 +249,10 @@ const QueryGenerator = {
     }
 
     if (this._dialect.supports.updateOnDuplicate && options.updateOnDuplicate) {
-      onDuplicateKeyUpdate += ' ON DUPLICATE KEY UPDATE ' + options.updateOnDuplicate.map(attr => {
-        const field = rawAttributes && rawAttributes[attr] && rawAttributes[attr].field || attr;
```
Comment (sushantdhiman) [inline lib/dialects/abstract/query-generator.js]: This should be enough, what do you observe here, `rawAttributes` empty?

Comment (sushantdhiman) [reply lib/dialects/abstract/query-generator.js]: Can you work on test case first? `rawAttributes` must not be undefined, we can change this method parameters OR pass `rawAttributes` from methods which don't pass it OR use `options.model` instead.

Let me know if you get stuck, I will handle this fix myself then :)

Comment (ManuRua) [reply lib/dialects/abstract/query-generator.js]: Hi! Yeah, i was working on test case but i have been travelling some days and i need to stop. I will push test in this week ;)

Comment (ManuRua) [reply lib/dialects/abstract/query-generator.js]: I'm executing integration tests with docker but your referenced test for updateOnDuplicate option doesn't appear in the launched collection. Any idea? Am i doing something wrong?
Also, when i launch integration tests, 255 passing and 201 failing...

**Comment by sushantdhiman** [comment]
I have worked on another PR which will fix this issue, thanks for your efforts https://github.com/sequelize/sequelize/pull/9162

**Comment by ManuRua** [comment]
Ok, but i think that you could have helped me and respond my previous questions to continue.

**Comment by kumail-raza** [comment]
@sushantdhiman this issue is not fixed kindly check, using version sequelize v5.8.12

**Pull request closed without merging**